packages/react-graphql/src/hooks/mutation.ts

```diff
@@ -0,0 +1,29 @@
+import {OperationVariables} from 'apollo-client';
+import {MutationOptions} from 'react-apollo';
+import {DocumentNode} from 'graphql-typed';
+import {useMemo} from 'react';
+
+import {MutationHookOptions, MutationHookResult} from './types';
+import {useApolloClient} from './apollo-context';
+
+export default function useMutation<Data = any, Variables = OperationVariables>(
+  mutation: DocumentNode<Data, Variables>,
+  options: MutationHookOptions<Data, Variables> = {},
+): MutationHookResult<Data, Variables> {
+  const {client: overrideClient, ...mutationOptions} = options;
+
+  const client = useApolloClient(overrideClient);
+
+  const runMutation = useMemo(
+    () => (perMutationOptions?: MutationOptions<Data, Variables>) => {
+      return client.mutate({
+        mutation,
+        ...mutationOptions,
+        ...perMutationOptions,
+      });
+    },
+    [client, mutation, mutationOptions],
```

Comment:
I am a bit worried about putting `mutationOptions` here, since it is pretty unlikely the consumer will know to memoize them. Maybe we should try to turn this into a set of primitives instead?